hare/broker.go

```diff
@@ -154,22 +118,40 @@ func (b *Broker) handleMessage(ctx context.Context, msg []byte) error {
 	logger.Debug("broker received hare message")
 
 	msgLayer := hareMsg.Layer
-	if !b.Synced(ctx, msgLayer) {
-		return errNotSynced
+
+	if !(b.nodeSyncState.IsSynced(ctx) && b.nodeSyncState.IsBeaconSynced(msgLayer.GetEpoch())) {
+		err := fmt.Errorf("broker rejecting message because not synced")
+		logger.With().Debug("rejecting message", log.Err(err))
+		return err
 	}
+	b.mu.Lock()
+	out, exist := b.outbox[msgLayer.Uint32()]
+	if !exist {
+		logger.With().Debug(
+			"broker received a message to a consensus process that is not registered",
+			msgLayer.Field(),
+			log.Stringer("msg_type", hareMsg.InnerMsg.Type),
+		)
 
-	isEarly := false
-	if err = b.validateTiming(ctx, &hareMsg); err != nil {
-		if !errors.Is(err, errEarlyMsg) {
-			// not early, validation failed
-			logger.With().Debug("broker received a message to a consensus process that is not registered",
-				log.Err(err))
-			return err
+		// We consider messages for the next layer to be early and we process
+		// them, all other messages we ignore. We don't impose the outbox
+		// capacity at this point since this is an externally initiated event
+		// and at most it can cause us to exceed our capacity by one slot.
+		if msgLayer == b.latestLayer.Add(1) {
+			out = make(chan any, inboxCapacity)
+			b.outbox[msgLayer.Uint32()] = out
+		} else {
+			b.mu.Unlock()
+			// This error is never inspected except to determine if it is not nil.
+			return fmt.Errorf(
+				"ignoring message to unregistered consensus process, latestLayer: %v, messageLayer: %v, messageType: %v",
```

Comment:
please try to shorten error msg here, should upstream caller decide to print it :)
maybe just `msg layer not registered`. and it's more consistent with the rest of the code to print round number instead of msg type (can be inferred by by round number)

Reply:
I'm not understanding the problem with having a long message here? As a caller if something goes wrong here the context of why it went wrong is useful to deciphering the problem. As a node operator or a dev debugging something I'd much rather have a long message that let me understand better what went wrong than a short message that doesn't help me understand the source of the problem.

Reply:
E.g: `ignoring message to unregistered consensus process, latestLayer: 0, messageLayer: 1237643, round: 0` would immediately let me know something was badly wrong with the sync or startup of the node. As opposed to `ignoring message to unregistered consensus process` which would leave me in a position where I need to start re-creating the issue, adding extra debug logs, maybe debugging ... etc. Before reaching the same understanding that was immediately apparent from the error.

Reply:
you can always grep for the exact text when you are debugging.... i don't see why long msg is different from short msg. the key here is the extra context (layer and round)